dev/top_printers.ml

```diff
@@ -60,19 +60,21 @@ let prrecarg = function
      str "Imbr[" ++ MutInd.print mind ++ pr_comma () ++ int i ++ str "]"
 let ppwf_paths x = pp (Rtree.pp_tree prrecarg x)
 
+let get_current_context = Vernacstate.Proof_global.get_current_context
```

Comment:
This is a non-blocking question: I believe that at some point you claimed the compat layer was needed only because STM could not be ported to the functional state. What's the longer term plan for these debug printers, then?

I'm just trying to see what will be the next steps to remove the compat layer.

Reply:
I guess that'd be up to the users of these interfaces, we could always have `Drop.` save a reference to the current printing state, however it would be the responsibility of the Drop user to manage such operations. The thing with Drop access is that you can always break things.

Reply:
For example we already do something like that, see `drop_last_doc`, so I guess we'd do `drop_last_state` too.

Reply:
On the other hand, I guess that debug printers such as the ones you want to install in the style of https://github.com/ocaml-community/utop/pull/269 should be "faithful", thus, don't take any env / sigma and print the raw object.